components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -473,13 +481,14 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
             return buildIntrospectionErrorResponse("Invalid access delegation");
         }
 
-        // Validate scopes.
+        // Validate scopes at app level.
         if (!tokenValidator.validateScope(messageContext)) {
             // This is redundant. But sake of readability.
             introResp.setActive(false);
-            return buildIntrospectionErrorResponse("Scope validation failed");
+            return buildIntrospectionErrorResponse("Scope validation failed at app level");
```

Comment:
Do we need to mention at which level the scope validation fails in the response? 
We could include that detail in a debug log.

@IsuraD WDYT?

Reply:
+1 to have a detailed debug log